src/Admin/VMware.ScriptRuntimeService.AdminEngine/RunspaceProviders/RunspaceProviders.cs

```diff
@@ -0,0 +1,62 @@
+// **************************************************************************
+//  Copyright 2020-2023 VMware, Inc.
+//  SPDX-License-Identifier: Apache-2.0
+// **************************************************************************
+
+
+using System;
+using System.Collections.Generic;
+using System.Text;
+using Microsoft.Extensions.Logging;
+using VMware.ScriptRuntimeService.AdminEngine.ConfigFileWriters;
+using VMware.ScriptRuntimeService.AdminEngine.VCRegistration;
+
+namespace VMware.ScriptRuntimeService.AdminEngine.RunspaceProviders {
+   public class RunspaceProviders {
+      private readonly ILoggerFactory _loggerFactory;
+      private readonly IConfigWriter _configWriter;
+      private readonly IConfigReader _configReader;
+      private readonly ILogger _logger;
+
+      public RunspaceProviders(ILoggerFactory loggerFactory, IConfigWriter configWriter, IConfigReader configReader) {
+         _loggerFactory = loggerFactory ?? throw new ArgumentNullException(nameof(loggerFactory));
+         _logger = loggerFactory.CreateLogger(typeof(VCRegistrator));
+         _configWriter = configWriter ?? throw new ArgumentNullException(nameof(configWriter));
+         _configReader = configReader ?? throw new ArgumentNullException(nameof(configReader));
+      }
+
+      public RunspaceProviderSettings UpdateSettings(
+         uint? numberOfRunspaces,
+         uint? runspaceIdleTimeMinutes,
+         uint? runspaceActiveTimeMinutes) {
+
+         dynamic data = _configReader.ReadSettings<dynamic>("service-settings", "settings.json");
+
+         if (numberOfRunspaces.HasValue && numberOfRunspaces.Value > 0) {
+            data.RunspaceProviderSettings.MaxNumberOfRunspaces = numberOfRunspaces.Value;
```

Comment:
What is the default value? Isn't 0 treated as unlimited? Applies to the other settings too. The way you implemented it seems like user cannot set it to 0 or unlimited.

Reply:
no, there is no unlimited logic